src/reactive-membrane.ts

```diff
@@ -35,15 +35,20 @@ export interface ObservableMembraneInit {
     valueObserved?: ReactiveMembraneAccessCallback;
     valueDistortion?: ReactiveMembraneDistortionCallback;
     valueIsObservable?: ReactiveMembraneObservableCallback;
+    magicPropertyKey?: PropertyKey;
 }
 
-function createShadowTarget(value: any): ReactiveMembraneShadowTarget {
+function createShadowTarget(value: any, magicPropertyKey: PropertyKey | undefined): ReactiveMembraneShadowTarget {
     let shadowTarget: ReactiveMembraneShadowTarget | undefined = undefined;
     if (isArray(value)) {
         shadowTarget = [];
     } else if (isObject(value)) {
         shadowTarget = {};
     }
+    if (!isUndefined(magicPropertyKey)) {
```

Comment:
these `!isUndefined` checks are weird. For the property key I'd suggest creating a method for: isPropertyKey following the [semantics of ES](https://tc39.es/ecma262/#sec-ispropertykey):

1. If Type(argument) is String, return true.
1. If Type(argument) is Symbol, return true.
1. Return false.

Reply:
the `isUndefined` is weird, yes... ideally we can have `isDefined` but I don't think TS allows to do a good declaration of that, @pmdartus might know how... but in general, we use isUndefined everywhere... see my previous message.

In this case, it is because the magical property key, which is well defined in the type system, could be undefined or a valid property key.